react/src/components/KeypairInfoModal.tsx

```diff
@@ -1,125 +1,130 @@
 
-interface KeypairInfoModalProps extends BAIModalProps {
+interface KeypairInfoModalProps extends ModalProps {
+  keypairInfoModalFrgmt: KeypairInfoModalFragment$key | null;
   onRequestClose: () => void;
 }
 
 const KeypairInfoModal: React.FC<KeypairInfoModalProps> = ({
+  keypairInfoModalFrgmt = null,
   onRequestClose,
-  ...baiModalProps
+  ...modalProps
 }) => {
-  const { t } = useTranslation();
-  const [userInfo] = useCurrentUserInfo();
-  const baiClient = useSuspendedBackendaiClient();
-  const { data: keypairs } = useTanQuery({
-    queryKey: ['baiClient.keypair.list', baiModalProps.open], // refetch on open state
-    queryFn: () => {
-      return baiModalProps.open
-        ? baiClient.keypair
-            .list(
-              userInfo.email,
-              ['access_key', 'secret_key', 'is_active'],
-              true,
-            )
-            .then((res: any) => res.keypairs)
-        : null;
-    },
-    staleTime: 0,
-  });
+  const { token } = theme.useToken();
 
-  const supportMainAccessKey = baiClient?.supports('main-access-key');
+  const keypair = useFragment(
+    graphql`
+      fragment KeypairInfoModalFragment on KeyPair {
+        user_id
+        access_key
+        secret_key
+        is_admin
+        created_at
+        last_used
+        resource_policy
+        num_queries
+        rate_limit
+        concurrency_used @since(version: "24.09.0")
+      }
+    `,
+    keypairInfoModalFrgmt,
+  );
 
+  // FIXME: Keypair query does not support main_access_key info.
   const { user } = useLazyLoadQuery<KeypairInfoModalQuery>(
     graphql`
-      query KeypairInfoModalQuery($email: String) {
-        user(email: $email) {
-          email
-          main_access_key @since(version: "23.09.7")
+      query KeypairInfoModalQuery($domain_name: String, $email: String) {
+        user(domain_name: $domain_name, email: $email) {
+          main_access_key @since(version: "24.03.0")
         }
       }
     `,
     {
-      email: userInfo.email,
+      email: keypair?.user_id,
+    },
+    {
+      fetchPolicy:
+        modalProps.open && keypair?.user_id ? 'network-only' : 'store-only',
     },
   );
 
   return (
     <BAIModal
-      {...baiModalProps}
-      title={t('usersettings.MyKeypairInfo')}
-      centered
-      onCancel={onRequestClose}
-      destroyOnClose
-      width={'auto'}
-      footer={[
-        <Button
-          key="keypairInfoClose"
-          onClick={() => {
-            onRequestClose();
-          }}
-        >
-          {t('button.Close')}
-        </Button>,
-      ]}
+      title={
+        <Flex gap={'xs'}>
+          <Typography.Text style={{ fontSize: token.fontSizeHeading5 }}>
+            {t('credential.KeypairDetail')}
+          </Typography.Text>
+          {user?.main_access_key === keypair?.access_key && (
+            <Tag color="red">{t('credential.MainAccessKey')}</Tag>
+          )}
+        </Flex>
+      }
+      onCancel={() => onRequestClose()}
```

Comment:
There is an OK button, but the onOK handler is missing. Since this modal is for informational purposes only, please modify it to include only a Close button instead of OK and Cancel buttons.

Reply:
Based on our discussion, I modified both `UserInfoModal` and `KeypairInfoModal` to not provide a footer.